### ISSUE TITLE
Recommend dependencies as SerVer by default

### ISSUE THREAD

**Issue by fulldecent** [rubygems/rubygems.org#3318]
_Alternate title: don't be pessimistic when recommending package versions_

## Is your feature request related to a problem?

No

## Describe the solution you'd like

We should promote SemVer dependencies for people browsing available packages on RubyGems.org.

This can be achieved simply by editing the text in this box.

<img width="1379" alt="Screenshot 2022-12-29 at 09 28 27" src="https://user-images.githubusercontent.com/382183/209967503-2a764409-976b-4a99-a537-c966128f1c0d.png">

## Describe alternatives you've considered

Create a new SemVer operator in Ruby -- https://github.com/rubygems/rubygems/issues/1919

## Additional context

In 2022 most packages are following SemVer release methodology.

By encouraging visitors of RubyGems.org to use the tighter "pessimistic" versioning, this means there is more chance of version conflicts in your project.

For example:

1. Your project requires foo with version >= 1.2.3 and ~ 1.2.3.
2. Your project also depends on bar, which requires foo with version >= 1.0.1 and ~ 1.0.1.

This means that no version of foo can satisfy your requirements.

However, foo version 1.2.3 probably exposes an API that is compatible with 1.0.1 because SemVar, so the version 1.2.3 /should/ be acceptable for you.

## Proposal

On this page:

https://rubygems.org/gems/lightning_sites

The "GEMFILE:" line should be 

```
gem 'lightning_sites', '>= 1.4.25', '< 2.0.0'
```

Instead of the current:

```
gem 'lightning_sites', '~> 1.4', '>= 1.4.25'
```

**Comment by simi** [comment]
First of all, I think we can restructure the page a little (add that gemfile block to the main part) to not make the most of the content hidden (and just copy unknown text).

As you mentioned, currently there is `gem 'lightning_sites', '~> 1.4', '>= 1.4.25'` suggested. That includes any `1.x` higher than `1.4.25` or same. How is `'>= 1.4.25', '< 2.0.0'` different? Which case is covered by your suggestion and not by current constraint?

:thinking: If I understand it well, in your example situation your suggested gem line would not make any difference.

---

Regarding `In 2022 most packages are following SemVer release methodology.`. Do you have any details on that?

---

In short, currently RubyGems doesn't bring any opinion on version constraints. It just follows what's specified in gemspec or in Gemfile (when using Bundler). When `bar` dependency is locked to `>= 1.0.1 and ~> 1.0.1` of `foo`, there is no way to allow `foo` `1.2.3` and that is done by design. Currently the way to fix this is to reach `bar` maintainer and ask to open dependency constraint if possible.

**Comment by fulldecent** [comment]
Hmm, you're right thank you.

I don't know where all the `~> x.y.z` dependencies came from in my projects. So I just assumed they came from RubyGems.org copy/pasta. But no, it looks like you said it is currently SemVer.

---

Good point, moving that to a new issue: https://github.com/rubygems/rubygems.org/issues/3323
